### PR TITLE
Add the braceexpand Python module.

### PR DESCRIPTION
**Pull request by cbrueffer**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/braceexpand/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% set name = "braceexpand" %}
+{% set version = "0.1.2" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "13980ad19c80ed6097c245febb694e57baffd8179b50bc80437003eb01465ecb"
```
Comment (marcelotrevisani) [inline recipes/braceexpand/meta.yaml]:
```suggestion
  sha256: 13980ad19c80ed6097c245febb694e57baffd8179b50bc80437003eb01465ecb
```

File: recipes/braceexpand/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
```
Comment (marcelotrevisani) [inline recipes/braceexpand/meta.yaml]:
```suggestion
  script: "{{ PYTHON }} -m pip install . -vv"
```

Comment (cbrueffer) [reply recipes/braceexpand/meta.yaml]: Thanks! According to https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip `--no-deps` should stay, any opinion?

Comment (marcelotrevisani) [reply recipes/braceexpand/meta.yaml]: https://github.com/conda-forge/staged-recipes#7-when-or-why-do-i-need-to-use--python---m-pip-install---vv

I think we found one doc not up to date

the ``"{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "`` is the "old" style

Comment (marcelotrevisani) [reply recipes/braceexpand/meta.yaml]: let the ``--no-deps`` then
it will not hurt
=p

**Pull request merged**